### PR TITLE
fix(UI): put legacy label on the old js curriculum title

### LINKED ISSUE #52902: Legacy JS Certification should be labeled with "Legacy" and moved on profile page
### Describe the Issue

The Legacy curriculum all have the "Legacy" label in front of them except the old JS curriculum:

<details>
<summary>Map</summary>

Screenshot 2024-01-02 at 7 41 02 PM

</details>

<details>
<summary>Superblock Page</summary>

Screenshot 2024-01-02 at 7 41 46 PM

</details>

<details>
<summary>Settings</summary>

Screenshot 2024-01-02 at 7 42 15 PM

</details>

<details>
<summary>Certification</summary>

Screenshot 2024-01-02 at 7 40 26 PM

</details>

In all the images above, it should say "Legacy JavaScript Algorithms and Data Structures"

Also, on the profile page, it should be moved to the Legacy area:

<details>
<summary>image</summary>

profile

</details>

### PR DESCRIPTION
**Pull request by mmatsumoto1026**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52902

This PR was opened to put the "Legacy" label in front of the old JS curriculum title in [maps](https://www.freecodecamp.org/learn/), [Superblock page](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/), [settings page](https://www.freecodecamp.org/settings#certification-settings), certification page, and profile page.

Then, the "View certification" button for the old JS will be moved to Legacy area on the profile page.

With using gitpod, the code was tested on windows 11 Home PC using Chrome, Firefox, Edge, and Brave browser.

### CODE REVIEW

**Comment by mmatsumoto1026** [comment]
Thanks for your information, @huyenltnguyen.

Therefore, resolving the conflicts was done.

**Comment by mmatsumoto1026** [comment]
It seems that there are still some errors in Playwright e2e test.

But after investigation, I couldn't figure out why the target or browser were closing even though the locator was able to retrieve the appropriate element in the tests, and why final output text test is flacky.

Solving these errors may be outside the scope of this pull request, so I'd like to leave the code as is for review.

**Review by huyenltnguyen — APPROVED** [review]
I believe the test failed because the super block links were clicked at the end of their assertions. The click caused the page to navigate to the superblock page, and the content of the page was no longer what we expect.

This should have already been the case in `main`, but I don't know why the test only failed here. Regardless, I removed the click simulation as it's not needed in the test. All links are clickable, and we already have an assertion to check their destination.

Moral lesson: Don't click the links.

---
Tested locally and verified that the old JS curriculum has been renamed to "Legacy".

<details>
<summary>Screenshots</summary>

| Page | Screenshot |
| --- | --- |
| /learn | Screenshot 2024-01-12 at 14 48 43 |
| //learn/javascript-algorithms-and-data-structures | Screenshot 2024-01-12 at 14 53 53 |
| /settings#certification-settings | Screenshot 2024-01-12 at 14 52 09 |
| /settings#certification-settings | Screenshot 2024-01-12 at 14 52 58 |
| /{username} | Screenshot 2024-01-12 at 14 54 40 |
| /certification/username}/javascript-algorithms-and-data-structures | Screenshot 2024-01-12 at 14 55 15 |
</details>

**Review by Sembauke — APPROVED** [review]

File: client/i18n/locales/english/translations.json [hunk 1]
```diff
@@ -912,9 +912,10 @@
     "title": {
       "Responsive Web Design": "Responsive Web Design",
       "responsive-web-design": "Responsive Web Design Certification",
-      "JavaScript Algorithms and Data Structures": "JavaScript Algorithms and Data Structures",
-      "javascript-algorithms-and-data-structures": "JavaScript Algorithms and Data Structures Certification",
+      "JavaScript Algorithms and Data Structures": "Legacy JavaScript Algorithms and Data Structures",
+      "javascript-algorithms-and-data-structures": "Legacy JavaScript Algorithms and Data Structures Certification",
       "javascript-algorithms-and-data-structures-v8": "JavaScript Algorithms and Data Structures Certification",
+
```
Comment (Sembauke) [inline client/i18n/locales/english/translations.json:918]:
```suggestion
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: client/i18n/locales/english/translations.json (hunk 1 above)
Comment (mmatsumoto1026) [reply client/i18n/locales/english/translations.json:918]: Sorry, I accidentally added an unnecessary line when resolving the conflict.

**Review by Sembauke — APPROVED** [review]

**Comment by mmatsumoto1026** [comment]
@huyenltnguyen Thanks for your comment.
After updating the branch, I've created the new PR.

**Pull request merged**